fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/AbstractValue.java

```diff
@@ -77,4 +89,22 @@ private int computeSemanticHashCode() {
         return fold(Value::hashCodeWithoutChildren,
                 (hashCodeWithoutChildren, childrenHashCodes) -> Objects.hash(childrenHashCodes, hashCodeWithoutChildren));
     }
+
+    @Override
+    public int height() {
+        return heightSupplier.get();
+    }
+
+    private int computeHeight() {
+        return StreamSupport.stream(getChildren().spliterator(), false).mapToInt(TreeLike::height).max().orElse(0) + 1;
```

Comment:
`Streams.stream(...)` please.

Reply:
Done. Method is refactored and sits now in `TreeLike`.